aptos-move/e2e-move-tests/src/tests/fee_payer.rs

```diff
@@ -74,6 +77,121 @@ fn test_account_not_exist_with_fee_payer_create_account() {
     assert!(bob_start > bob_after);
 }
 
+#[test]
+fn test_account_not_exist_and_out_of_gas_with_fee_payer_create_account() {
+    // This should result in an aborted, yet committed transaction and an account created.
+    let mut h = MoveHarness::new_with_features(vec![FeatureFlag::GAS_PAYER_ENABLED], vec![]);
+
+    let alice = Account::new();
+    let bob = h.new_account_at(AccountAddress::from_hex_literal("0xb0b").unwrap());
+
+    let alice_start =
+        h.read_resource::<CoinStoreResource>(alice.address(), CoinStoreResource::struct_tag());
+    assert!(alice_start.is_none());
+    let bob_start = h.read_aptos_balance(bob.address());
+
+    let payload = aptos_stdlib::aptos_coin_transfer(*alice.address(), 1);
+    let transaction = TransactionBuilder::new(alice.clone())
+        .fee_payer(bob.clone())
+        .payload(payload)
+        .sequence_number(0)
+        .max_gas_amount(99_999) // This is not enough to execute this transaction
+        .gas_unit_price(1)
+        .sign_fee_payer();
+
+    let output = h.run_raw(transaction);
+    assert!(transaction_status_eq(
+        output.status(),
+        &TransactionStatus::Discard(StatusCode::MAX_GAS_UNITS_BELOW_MIN_TRANSACTION_GAS_UNITS)
+    ));
+
+    let alice_after =
+        h.read_resource::<CoinStoreResource>(alice.address(), CoinStoreResource::struct_tag());
+    assert!(alice_after.is_none());
+    let bob_after = h.read_aptos_balance(bob.address());
+    assert_eq!(bob_start, bob_after);
+}
+
+#[test]
+fn test_account_not_exist_and_abort_with_fee_payer_create_account() {
+    // This should result in an aborted, yet committed transaction and an account created.
+    let mut h = MoveHarness::new_with_features(vec![FeatureFlag::GAS_PAYER_ENABLED], vec![]);
+
+    let alice = Account::new();
+    let bob = h.new_account_at(AccountAddress::from_hex_literal("0xb0b").unwrap());
+
+    let alice_start =
+        h.read_resource::<CoinStoreResource>(alice.address(), CoinStoreResource::struct_tag());
+    assert!(alice_start.is_none());
+    let bob_start = h.read_aptos_balance(bob.address());
+
+    let payload = aptos_stdlib::aptos_coin_transfer(*alice.address(), 1);
+    // Offered max fee is 100K, the minimum to execute this transaction
+    let transaction = TransactionBuilder::new(alice.clone())
+        .fee_payer(bob.clone())
+        .payload(payload)
+        .sequence_number(0)
+        .max_gas_amount(50_000)
+        .gas_unit_price(2)
+        .sign_fee_payer();
+
+    let output = h.run_raw(transaction);
+    // ECOIN_STORE_NOT_PUBLISHED
```

Comment:
Where does the error come from? I think this can be returned from both `coin::deposit` from transfer and `coin::balance` which is used in prologue. Can we have a test which results in Move abort guaranteed to be inside a txn, e.g. division by zero?

Reply:
When we run a transaction, we split into prologue, user txn, epilogue, right? The error you are checking here comes from prologue, I believe, so you are not testing the user txn execution path. Adding a test which does divide by zero allows us to test the behaviour when failure comes inside of a user txn.

Reply:
This is failing during user txn. A sponsored transaction sending account does not need coin store. This attempts to access a coin store only during user txn. How to be sure? We were charged gas. That's only possible for transactions that succeed prologue. I do not think adding a more complex test improves testing. Maybe we have something I can copy over, but this test also demonstrates how powerful this concept is. It can reduce account costs for certain applications by 50%.

Reply:
if one exists, happy to use it, but the complexity is if we add writing more one-off code for testing